src/main/java/org/lwes/emitter/EmitterGroupBuilder.java

```diff
@@ -58,223 +75,17 @@
-    }
-  }
 
   public static EmitterGroup createGroup(Properties props, String groupName,
                                          EventFactory factory)
       throws IOException {
     String prefix = "lwes." + groupName + ".";
-
-    String defaultPortStr = props.getProperty(prefix + "port");
-    int defaultPort = (defaultPortStr != null) ? Integer.parseInt(defaultPortStr) : -1;
-
-    String hostsStr = props.getProperty(prefix + "hosts");
-    String strategy = props.getProperty(prefix + "strategy");
-    boolean emitHeartbeat =
-      Boolean.parseBoolean(props.getProperty(prefix + "emit_heartbeat"));
-
-    String rateStr = props.getProperty(prefix + "sample_rate");
-    double defaultSampleRate = null == rateStr ? 1.0 : Double.parseDouble(rateStr);
-
-    EmitterGroupFilter filter = getEmitterGroupFilter(props, prefix);
-    if (filter != null) {
-      LOG.info(String.format("Emitter group %s : %s", groupName, filter));
-    }
-
-    if (STRATEGY_NESTED.matcher(strategy).matches()) {
-      return buildNestedEmitterGroup(prefix, hostsStr, strategy, defaultPort, filter, defaultSampleRate, emitHeartbeat, factory);
-    }
-
-    DatagramSocketEventEmitter<?>[] emitters =
-      createEmitters(groupName, prefix, hostsStr, defaultPort, emitHeartbeat, factory);
-
-    if (strategy == null || strategy.isEmpty()) {
-      throw new RuntimeException(
-          String.format(
-              "No strategy specified for emitter group %s - not set in %s property",
-              groupName,
-              prefix + "strategy"));
-    }
-
-    Matcher mOfN = STRATEGY_M_OF_N.matcher(strategy);
-
-    if (STRATEGY_ALL.equalsIgnoreCase(strategy)) {
-      return new BroadcastEmitterGroup(emitters, filter, defaultSampleRate, factory);
-    } else if (mOfN.matches()) {
-      return new MOfNEmitterGroup(emitters, Integer.parseInt(mOfN.group(1)), filter, defaultSampleRate, factory);
-    } else {
-      throw new RuntimeException(
-          String.format(
-              "Invalid strategy '%s' given for emitter group %s in property %s.",
-              strategy,
-              groupName,
-              prefix + "strategy"));
-    }
-  }
-
-  public static EmitterGroup createGroup(Properties props, String groupName) throws IOException {
-    return createGroup(props, groupName, null);
-  }
-
-  private static EmitterGroup buildNestedEmitterGroup(String prefix, String hostsStr, String strategyStr, int port, EmitterGroupFilter filter, double sampleRate, boolean emitHeartbeat, EventFactory factory) throws IOException {
-  String[] ratioConfig = strategyStr.split("_");
-  if (null == ratioConfig || ratioConfig.length != 2) {
-    throw new IllegalArgumentException("Invalid nested strategy config " + strategyStr);
-  }
-  if (!nestedTuplePattern.matcher(hostsStr).find()) {
-    throw new IllegalArgumentException("Invalid nested hosts config " + hostsStr);
+    String type = props.getProperty(prefix + "type", "udp");
```

Comment:
Comment added for clarity - feel free to reject.
```suggestion
    // Default to UDP type for backwards compatibility
    String type = props.getProperty(prefix + "type", "udp");
```